handling-null-values-in-avro-and-protobuf/kafka/README.md

```diff
@@ -0,0 +1,231 @@
+
+#### Run the commands
+
+[Sign up](https://www.confluent.io/) for a Confluent Cloud account if you haven't already. 
+
+Login, and then click 'Environments -> Create Cloud Environment' and create a cloud environment using the defaults there. 
+
+Navigate to your environment and click 'Add cluster'. Create a cluster using the default values provided. 
+
+Click 'Topics -> Add topic' to create two topics with the default values, one named 'avro-purchase' and the other 'proto-purchase' (we'll cover null values in Protobuf schemas later in the tutorial). 
+
+On the right-hand navbar, click 'API keys -> Add key -> Global access'. Download the values as you will need them to run this tutorial. 
+
+In the same navbar, click 'Clients -> Choose Your Language -> Java -> Create Schema Registry API key'. Save this key and secret as well as the URL listed in the configuration snippet. 
+
+Create a folder anywhere you like to house the code this tutorial:
+
+```bach
+mkdir handling-null-values
+```
+
+Now, create a file at `handling-null-values/resources/confluent.properties` with these values in it:
+
+```
+# Required connection configs for Kafka producer, consumer, and admin
+bootstrap.servers=BOOTSTRAP_URL/S
+security.protocol=SASL_SSL
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username='USERNAME' password='PASSWORD';
+sasl.mechanism=PLAIN
+use.latest.version=true
+# Required for correctness in Apache Kafka clients prior to 2.6
+client.dns.lookup=use_all_dns_ips
+
+wrapper.for.nullables=true
+key.converter=io.confluent.connect.avro.AvroConverter
+key.converter.schema.registry.url=SR_URL/S
+value.converter=io.confluent.connect.avro.AvroConverter
+value.converter.schema.registry.url=CONVERTER_SR_URL/S
+
+# Best practice for higher availability in Apache Kafka clients prior to 3.0
+session.timeout.ms=45000
+
+# Best practice for Kafka producer to prevent data loss
+acks=all
+
+# Required connection configs for Confluent Cloud Schema Registry
+schema.registry.url=SR_URL/S
+basic.auth.credentials.source=USER_INFO
+basic.auth.user.info=API_KEY:SECRET
+```
+
+Replace the USERNAME and PASSWORD values with the Confluent Cloud key and secret respectively. Add the url from the schema registry client configuration snippet for `SR_URL/S` and add the schema registry API key and secret for `basic.auth.user.info`, retaining the colon in the placeholder. 
+
+Inside `handling-null-values/kafka/code/src/main/avro/purchase.avsc` you'll see: 
+
+```
+{
+  "type":"record",
+  "namespace": "io.confluent.developer.avro",
+  "name":"PurchaseAvro",
+  "fields": [
+    {"name": "item", "type": ["string", "null"] },
+    {"name": "total_cost", "type": "double" },
+    {"name": "customer_id", "type": "string"}
+  ]
+}
+```
+
+When you run `gradle runAvroProducer` and furthermore, `gradle runAvroConsumer`, you'll see that the events with null items are produced and consumed successfully. 
```

Comment:
use the top level gradle wrapper so gradle doesnt have to be a dependency. from the top level I think it'd be:

```
./gradlew :handling-null-values-in-avro-and-protobuf:kafka:runAvroProducer
```

the same applies to the `gradle` commands below for the protobuf example